utilities/test_suite/HIP/Tensor_hip.cpp

```diff
@@ -0,0 +1,1877 @@
+    {
+        Rpp8u *inputTemp, *inputSecondTemp;
+        Rpp32f *inputf32Temp, *inputf32SecondTemp;
+        inputTemp = inputu8 + srcDescPtr->offsetInBytes;
+        inputSecondTemp = inputu8Second + srcDescPtr->offsetInBytes;
+        inputf32Temp = (Rpp32f *)((Rpp8u *)input + srcDescPtr->offsetInBytes);
+        inputf32SecondTemp = (Rpp32f *)((Rpp8u *)input + srcDescPtr->offsetInBytes);
+
+        for (int i = 0; i < ioBufferSize; i++)
+        {
+            *inputf32Temp = ((Rpp32f)*inputTemp) * conversionFactor;
+            *inputf32SecondTemp = ((Rpp32f)*inputSecondTemp) * conversionFactor;
+            inputTemp++;
+            inputf32Temp++;
+            inputSecondTemp++;
+            inputf32SecondTemp++;
+        }
+    }
+    else if (inputBitDepth == 5)
+    {
+        Rpp8u *inputTemp, *inputSecondTemp;
+        Rpp8s *inputi8Temp, *inputi8SecondTemp;
+
+        inputTemp = inputu8 + srcDescPtr->offsetInBytes;
+        inputSecondTemp = inputu8Second + srcDescPtr->offsetInBytes;
+        inputi8Temp = (Rpp8s *)input + srcDescPtr->offsetInBytes;
+        inputi8SecondTemp = (Rpp8s *)input_second + srcDescPtr->offsetInBytes;
+
+        for (int i = 0; i < ioBufferSize; i++)
+        {
+            *inputi8Temp = (Rpp8s) (((Rpp32s) *inputTemp) - 128);
+            *inputi8SecondTemp = (Rpp8s) (((Rpp32s) *inputSecondTemp) - 128);
+            inputTemp++;
+            inputi8Temp++;
+            inputSecondTemp++;
+            inputi8SecondTemp++;
+        }
+    }
+
+    hipMalloc(&d_input, inputBufferSize);
+    hipMalloc(&d_input_second, inputBufferSize);
+    hipMalloc(&d_output, outputBufferSize);
+    hipMemcpy(d_input, input, inputBufferSize, hipMemcpyHostToDevice);
+    hipMemcpy(d_input_second, input_second, inputBufferSize, hipMemcpyHostToDevice);
+    hipMemcpy(d_output, output, outputBufferSize, hipMemcpyHostToDevice);
+
+    // Run case-wise RPP API and measure time
+    rppHandle_t handle;
+    hipStream_t stream;
+    hipStreamCreate(&stream);
+    rppCreateWithStreamAndBatchSize(&handle, stream, noOfImages);
+
+    clock_t start, end;
+    double maxTimeUsed = 0, minTimeUsed = 500, avgTimeUsed = 0;
+    double gpuTime, wallTime;
+    string testCaseName;
+
+    // case-wise RPP API and measure time script for Unit and Performance test
+    printf("\nRunning %s %d times (each time with a batch size of %d images) and computing mean statistics...", func.c_str(), numIterations, noOfImages);
+    for (int perfRunCount = 0; perfRunCount < numIterations; perfRunCount++)
+    {
+        clock_t start, end;
+        double startOmp, endOmp;
+        switch (testCase)
+        {
+        case 0:
+        {
+            testCaseName = "brightness";
+
+            Rpp32f alpha[images];
+            Rpp32f beta[images];
+            for (i = 0; i < images; i++)
+            {
+                alpha[i] = 1.75;
+                beta[i] = 50;
+            }
+
+            // Uncomment to run test case with an xywhROI override
```

Comment:
Lets modularize this XYWH/LTRB option, its almost the same on every function.

Reply:
Done